src/gpuarray_collectives_cuda_nccl.c

```diff
@@ -175,11 +175,14 @@ static inline ncclDataType_t convert_data_type(int typecode) {
   switch (typecode) {
   case GA_BYTE: return ncclChar;
   case GA_INT: return ncclInt;
-  case GA_HALF: return ncclHalf;
   case GA_FLOAT: return ncclFloat;
   case GA_DOUBLE: return ncclDouble;
   case GA_LONG: return ncclInt64;
   case GA_ULONG: return ncclUint64;
+  #ifdef CUDA_HAS_HALF
```

Comment:
I'm not sure what to do if CUDA_HAS_HALF isn't defined. I think it should always be defined as we only support recent enough cuda version. So I would just remove the ifdef. Do you agree with that?

otherwise, it look good. thanks.